superset/assets/src/middleware/loggerMiddleware.js

```diff
@@ -107,6 +107,7 @@ const loggerMiddleware = store => next => action => {
       ...eventData,
       event_type: 'user',
       event_id: lastEventId,
+      visibility: document.visibilityState,
```

Comment:
should we move this into the top level `eventData` so that it's added on timing events too?

Reply:
`timing event` (mostly is data loading and chart render) is triggered by `user event` (dashboard mount, switch tab, etc). I assume during 1 user event (a few charts will be loading and re-render), the browser's visibilityState will not change.